bundles/org.openhab.binding.asuswrt/src/main/java/org/openhab/binding/asuswrt/internal/helpers/AsuswrtUtils.java

```diff
@@ -0,0 +1,464 @@
+    /**
+     * Limit Value between limits
+     * 
+     * @param value Integer
+     * @param lowerLimit
+     * @param upperLimit
+     * @return limited value
+     */
+    public static Integer limitVal(@Nullable Integer value, Integer lowerLimit, Integer upperLimit) {
+        if (value == null || value < lowerLimit) {
+            return lowerLimit;
+        } else if (value > upperLimit) {
+            return upperLimit;
+        }
+        return value;
+    }
+
+    /************************************
+     * FORMAT UTILS
+     ***********************************/
+    /**
+     * return value or default val if it's null
+     * 
+     * @param <T> Type of value
+     * @param value value
+     * @param defaultValue defaut value
+     * @return value or default
+     */
+    public static <T> T getValueOrDefault(@Nullable T value, T defaultValue) {
+        return value == null ? defaultValue : value;
+    }
+
+    /**
+     * Format MAC-Address replacing old division chars and add new one
+     * 
+     * @param mac unformated mac-Address
+     * @param newDivisionChar new division char (e.g. ":","-" )
+     * @return new formated mac-Address
+     */
+    public static String formatMac(String mac, char newDivisionChar) {
+        String unformatedMac = unformatMac(mac);
+        String formatedMac = "";
+        try {
+            formatedMac = unformatedMac.replaceAll("(.{2})", "$1" + newDivisionChar).substring(0, 17);
+            return formatedMac;
+        } catch (Exception e) {
+            return mac;
+        }
+    }
+
+    /**
+     * unformat MAC-Address replace all division chars
+     */
+    public static String unformatMac(String mac) {
+        mac = mac.replace("-", "");
+        mac = mac.replace(":", "");
+        mac = mac.replace(".", "");
+        mac = mac.replace(" ", "");
+        return mac;
+    }
+
+    /**
+     * check if isvalid MAC-Address
+     */
+    public static boolean isValidMacAddress(String mac) {
+        // MAC-Addresses usually are 6 * 2 hex nibbles separated by colons,
+        // but apparently it is legal to have 4 * 3 hex nibbles as well,
+        // and the separators can be any of : or - or . or nothing.
+        return (patternMacPairs.matcher(mac).find() || patternMacTriples.matcher(mac).find());
+    }
+
+    /**
+     * HEX-STRING to byte convertion
+     */
+    public static byte[] hexStringToByteArray(String s) {
+        int len = s.length();
+        byte[] data = new byte[len / 2];
+        try {
+            for (int i = 0; i < len; i += 2) {
+                data[i / 2] = (byte) ((Character.digit(s.charAt(i), 16) << 4) + Character.digit(s.charAt(i + 1), 16));
+            }
+        } catch (Exception e) {
+        }
+        return data;
+    }
+
+    /**
+     * Return Boolean from string
+     * 
+     * @param s - string to be converted
+     * @param defVal - Default Value
+     */
+    public static Boolean stringToBool(@Nullable String s, boolean defVal) {
```

Comment:
You can also use the primitive because it is non-null.

```suggestion
    public static boolean stringToBool(@Nullable String s, boolean defVal) {
```